web/controllers/game_controller.ex

```diff
@@ -0,0 +1,11 @@
+defmodule Yggdrasil.GameController do
+  use Yggdrasil.Web, :controller
+
+  alias Yggdrasil.Game
+
+  def index(conn, _params) do
+    games = Repo.all Game
```

Comment:
OK so this example of context specific to me is confusing the web with the game. That's what I'm trying to avoid.

So ok one I would never want APIS like this and this may be your concern

```elixir
Yggdrasil.remove_logged_in_character()
Yggdrasil.remove_non_logged_in_character()
```

They are too specific. I don't think there is any value in taking something like remove_logged_in_character() and have it hide calls to check if a character is online and act differently if it is, or error differently. Totally agree the web knows best here. That is very context specific and indeed would be a pass through layer that would have to change all the time.

However on the flip side of that coin I certainly don't want a web controller knowing all the internals of a the game because for example how a character is determined to be online is something the game is best suited to answer, and how to shutdown a player process for that character is something that the game is best suited to answer and would know best.

Also how to remove a character from a game I think is best answered in the game app, it knows best how the game uses the character right? The game may require that in order to remove a character it can't be running, therefore it just simply doesn't allow it. The web can then say ok, I need to first shutdown the player process. 

These are things we will wrap up in modules anyway I believe and will be in Yggdrasil. 

So in your example with the context I full expect something like this

```Elixir
# admin removing a character
if Yggdrasil.Player.online(character_id) do
  :ok = Yggdrasil.Player.shutdown(character_id)
end
Yggdrasil.Character.remove(character_id)

# User trying to do it
case Yggdrasil.Player.online(character_id) do
  true -> # notify user they can't
  false -> Yggdrasil.Character.remove(character_id)
end
```

Otherwise this code would balloon up with having to know how to figure out all the stuff involved with characters... I don't quite see why would would want that or the benefit to it. I may have the wrong perspective here. But for example if we didn't user Player module here the web would need to know it's an ETS and talk to the ETS table.

Also in my example there, there it's probably best that we don't have like one monolithic API, I'm more or less full expecting lots of modules from Yggdrasil to be used and the web can orchestrate it. 

I'm just not a fan of direct Repo access unless it's needed, because then the web has to know too much I think.